3rdparty/lua-5.1.3/src/lvm.c

```diff
@@ -515,7 +515,24 @@ void luaV_execute (lua_State *L, int nexeccalls) {
             break;
           }
           case LUA_TSTRING: {
-            setnvalue(ra, cast_num(tsvalue(rb)->len));
+            const char *bytes = getstr(tsvalue(rb));
+            size_t index = 0;
+            size_t characterLength = 0;
+            unsigned char byte = bytes[index];
+            while (byte) {
+              size_t byteCount = 1;
+              if (byte >= 240) {
```

Comment:
Disclaimer: I'm not super familiar with UTF-8 encodings, so all this info is pulled from the [wiki entry](https://en.wikipedia.org/wiki/UTF-8)

With that in mind and according to the table on that page, there are a few restricted starting bytes that are always invalid and some starting bytes that are valid _some_ of the time, but not all the time. Not sure if we should be more specific in this if-else block or not to accommodate that.

That said, ahu tests in `wonder_content` seem to be passing, so I'm not sure what to make of it. In any case, perhaps we can move these comparison numbers into constants in the file.

Reply:
It's a good question. I don't know the answer yet. Also, this code assumes a valid UTF-8 encoding. That's probably a safe assumption for the data we control but not for anything provided by the user. Note the above code is for the string length `#` operator.

Reply:
Updated string length operator to be more strict.